src/MCPClient/lib/clientScripts/archivematicaCreateMETSReingest.py

```diff
@@ -0,0 +1,435 @@
+
+    return root
+
+
+def add_rights_elements(rights_list, amdsecs, now, rights_counter, updated=False):
+    """
+    Create and add rightsMDs for everything in rights_list to amdsecs.
+    """
+    # Add to files' amdSecs
+    for amdsec in amdsecs:
+        # Get element to add rightsMDs after
+        try:
+            # Add after other rightsMDs
+            add_after = amdsec.findall('mets:rightsMD', namespaces=ns.NSMAP)[-1]
+        except IndexError:
+            # If no rightsMDs, then techMD is aways there and previous subsection
+            add_after = amdsec.findall('mets:techMD', namespaces=ns.NSMAP)[-1]
+        for rights in rights_list:
+            # Generate ID based on number of other rightsMDs
+            rights_counter += 1
+            rightsid = 'rightsMD_%s' % rights_counter
+            print 'Adding rightsMD', rightsid,
+
+            # Get file UUID for this file
+            file_uuid = amdsec.findtext('mets:techMD/mets:mdWrap[@MDTYPE="PREMIS:OBJECT"]//premis:objectIdentifierValue', namespaces=ns.NSMAP)
+            print 'for file', file_uuid, 'with ADMID', amdsec.get('ID')
+
+            # Create element
+            rightsMD = etree.Element(ns.metsBNS + "rightsMD", ID=rightsid, CREATED=now)
+            mdWrap = etree.SubElement(rightsMD, ns.metsBNS + 'mdWrap', MDTYPE='PREMIS:RIGHTS')
+            xmlData = etree.SubElement(mdWrap, ns.metsBNS + 'xmlData')
+            rights_statement = createmetsrights.createRightsStatement(rights, file_uuid)
+            xmlData.append(rights_statement)
+
+            if updated:
+                rightsMD.set('STATUS', 'current')
+                # Find superseded rightsMD and mark as such
+                # rightsBasis is semantically unique (though not currently
+                # enforced in code). Find rightsMDs with the same rights basis
+                # and mark superseded
+                superseded = amdsec.xpath('mets:rightsMD[not(@STATUS) or @STATUS="current"]//premis:rightsBasis[text()="' + rights.rightsbasis + '"]/ancestor::mets:rightsMD', namespaces=ns.NSMAP)
+                for elem in superseded:
+                    print 'Marking', elem.get('ID'), 'as superseded'
+                    elem.set('STATUS', 'superseded')
+
+            add_after.addnext(rightsMD)
+            add_after = rightsMD
+
+    return rights_counter
+
+
+def add_events(root, sip_uuid):
+    """
+    Add reingest events for all existing files.
+    """
+    # Get all reingestion events for files in this SIP
+    reingest_events = models.Event.objects.filter(file_uuid__sip__uuid=sip_uuid, event_type='reingestion')
+    digiprov_counter = int(root.xpath('count(mets:amdSec/mets:digiprovMD)', namespaces=ns.NSMAP))  # HACK
+    # Get Agent
+    try:
+        agent = models.Agent.objects.get(identifiertype="preservation system", name="Archivematica", agenttype="software")
+    except models.Agent.DoesNotExist, models.Agent.MultipleObjectsReturned:
```

Comment:
Do we want to log the MultipleObjectsReturned case? That seems improbable enough we'd want to be aware of it.